### PR TITLE
Support custom preview for uploader component

### LINKED ISSUE #1203: Support custom preview for uploader component
# Feature request 🚀

### Expected:

- No breaking changes

### Examples:

```vue
<script setup>
import { ref } from 'vue'

const files = ref([])

function handlePreview(file) {
   // do something
}
</script>

<template>
  <var-uploader v-model="files" prevent-default-preview @preview="handlePreview" />
</template>
```

### Programme:
Prevent default preview behavior when setting `props.preventDefaultPreview = true` and listen to preview events to handle the preview process.

### PR DESCRIPTION
**Pull request by tiny-dust**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

Added props preventDefaultPreview and event onPreview to support customized file preview

## Issues

close #1203 .

### CODE REVIEW

File: packages/varlet-ui/src/uploader/Uploader.vue [hunk 1]
```diff
@@ -181,15 +181,20 @@ export default defineComponent({
     )
 
     function preview(varFile: VarFile) {
-      const { disabled, previewed } = props
+      const { disabled, previewed, preventDefaultPreview, onPreview } = props
+
+      if (preventDefaultPreview) {
+        call(onPreview, reactive(varFile))
+        return
+      }
```
Comment (haoziqaq) [inline packages/varlet-ui/src/uploader/Uploader.vue:189]: onPreview should always be called

File: packages/varlet-ui/src/uploader/Uploader.vue [hunk 2]
```diff
@@ -181,15 +181,20 @@ export default defineComponent({
...
 
       if (form?.disabled.value || disabled || !previewed) {
         return
       }
 
       const { url } = varFile
 
-      if (isHTMLSupportImage(url)) {
+      if (url && isHTMLSupportImage(url)) {
```
Comment (haoziqaq) [inline packages/varlet-ui/src/uploader/Uploader.vue:197]: don't need to check the url, isHTMLSupportImage already does that.

Comment (tiny-dust) [reply packages/varlet-ui/src/uploader/Uploader.vue:197]: ok~

File: packages/varlet-ui/src/uploader/__tests__/index.spec.js [hunk 3]
```diff
@@ -119,6 +119,26 @@ test('test uploader preview', async () => {
...
+      onPreview,
+    },
+  })
+
+  await wrapper.find('.var-uploader__file').trigger('click')
+  expect(onPreview).toHaveBeenCalledTimes(1)
+
+  wrapper.setProps({ preventDefaultPreview: false })
+  await wrapper.find('.var-uploader__file').trigger('click')
+  expect(onPreview).toHaveBeenCalledTimes(2)
```
Comment (haoziqaq) [inline packages/varlet-ui/src/uploader/__tests__/index.spec.js:139]: expect default preview behavior should be prevented.

File: packages/varlet-ui/src/uploader/docs/en-US.md [hunk 4]
```diff
@@ -47,6 +47,35 @@ const files = ref([
...
+import { Dialog } from '@varlet/ui'
+
+const files = ref([
+  {
+    url: 'https://varlet.gitee.io/varlet-ui/cat.jpg',
+    cover: 'https://varlet.gitee.io/varlet-ui/cat.jpg'
+  }
+])
+
+function handlePreview(files) {
```
Comment (haoziqaq) [inline packages/varlet-ui/src/uploader/docs/en-US.md:65]: files -> file

File: packages/varlet-ui/src/uploader/docs/zh-CN.md [hunk 5]
```diff
@@ -377,6 +406,8 @@ const files = ref([
 | `resolve-type` | 文件预处理类型，可选值为 `default` `file` `data-url`（`default`，图片文件包含 dataURL 编码和 File 对象，其他类型仅包含 File 对象。`file`，仅包含 File 对象。`data-url`，所有文件类型都包含 dataURL 编码和 File 对象） | _string_ | `default` |
 | `validate-trigger` | 触发验证的时机， 可选值为 `onChange` `onRemove` | _ValidateTriggers[]_ | `['onChange', 'onRemove']` |
 | `rules` | 验证规则，返回 `true` 表示验证通过，其余的值则转换为文本作为用户提示 | _Array<(v: VarFile, u: VarFileUtils) => any>_ | `-` |
+| `preventDefaultPreview` | 阻止默认预览行为 | _boolean_ | `false` |
```
Comment (haoziqaq) [inline packages/varlet-ui/src/uploader/docs/zh-CN.md:409]: typo

File: packages/varlet-ui/src/uploader/docs/zh-CN.md [hunk 6]
```diff
@@ -421,6 +452,8 @@ const files = ref([
 | `oversize` | 文件超过限制大小时触发 | `file: VarFile` |
 | `before-remove` | 文件删除前触发，返回假值阻止文件删除(支持 promise) | `file: VarFile` |
 | `remove` | 文件删除时触发 | `file: VarFile` |
+| `preview` | 文件预览时触发，可自定义预览方式，需要配合`preventDefaultPreview`参数阻止默认预览行为 | `file: VarFile` |
```
Comment (haoziqaq) [inline packages/varlet-ui/src/uploader/docs/zh-CN.md:455]: 删除"可自定义预览方式，需要配合`preventDefaultPreview`参数阻止默认预览行为"

**Review by haoziqaq — COMMENTED** [review]
please fix types/uploader.d.ts

**Pull request merged**